rust/lance-io/src/object_store/gcs_wrapper.rs

```diff
@@ -297,6 +328,20 @@ impl Upload {
     }
 }
 
+#[derive(Debug)]
+struct ConnectionResetError {
+    message: String,
+    source: Box<dyn std::error::Error + Send + Sync>,
+}
+
+impl std::error::Error for ConnectionResetError {}
+
+impl std::fmt::Display for ConnectionResetError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}: {}", self.message, self.source)
+    }
+}
```

Comment:
Why the dedicated error type?

Reply:
I want to make it clear that we did do a retry on this error. Otherwise, if we return the underlying error is saying zero retries.